src/calculation.rs

```diff
@@ -74,33 +77,36 @@ pub fn compile_and_write_results(
             ),
         );
     }
-    let result_filename = if abi_mode {
-        "./results/abi_results.json".to_string()
-    } else {
-        "./results/wasm_results.json".to_string()
+    let result_filename = match abi_type {
+        AbisType::AS => "./results/abi_results.json".to_string(),
+        AbisType::WasmV1 => "./results/wasm_results.json".to_string(),
     };
-    let mut output = File::create(result_filename).unwrap();
+
+    let mut output = File::create(&result_filename).unwrap_or_else(|_| {
+        panic!("Failed to create file {}", result_filename)
+    });
     write!(
         output,
         "{}",
         serde_json::to_string_pretty(&final_results).unwrap()
     )
     .unwrap();
+
     for (key, value) in final_results.iter() {
         gas_costs.insert(
-            if abi_mode {
-                format_key(key)
-            } else {
-                key.clone()
+            match abi_type {
+                AbisType::AS => format_key(key),
+                AbisType::WasmV1 => key.clone(),
             },
-            (max_gas as f64 / (max_execution_time.as_millis() as f64 / value.0)) as u32,
+            (max_gas as f64 / (max_execution_time.as_millis() as f64 / value.0))
+                as u32,
         );
     }
-    let output_filename = if abi_mode {
-        "./results/abi_gas_costs.json".to_string()
-    } else {
-        "./results/wasm_gas_costs.json".to_string()
+    let output_filename = match abi_type {
+        AbisType::AS => "./results/abi_gas_costs.json".to_string(),
+        AbisType::WasmV1 => "./results/wasm_gas_costs.json".to_string(),
```

Comment:
If I understand correctly, both AbisType should be in the same result file at the end no? wasm_gas_costs was the name of the file used for the gas of WASM operations (in the WAT SC), not wasmv1 abis

Reply:
I went a bit to fast, will fix this

Reply:
please see 86bfb76 that should fix this

Reply:
Looks good to me!